backend/flask_main.py

```diff
@@ -74,43 +68,43 @@ def login():
     if password_ok:
         login_user(user)
         session.permanent = True
-        return {'auth': 'ok'}  # , 'exp': }
+        return {'auth': 'ok'}
     return not_allowed_return_val
 
 
 @app.route('/logout')
 @public_endpoint
-def logout():
+def logout() -> FLASK_RESPONSE_TYPE:
     logout_user()
     return ''
 
 
 @app.route('/is_authenticated')
 @public_endpoint
-def is_authenticated():
+def is_authenticated() -> FLASK_RESPONSE_TYPE:
     if current_user.is_authenticated(session):
         return '', HTTP_STATUS_CODE.HTTP_200_OK.value
     return '', HTTP_STATUS_CODE.HTTP_401_UNAUTHORIZED.value
 
 
 @app.route('/current_users', methods=['GET', 'POST',])
-def current_users():
-    print(f'current_users {request.method}')
+def current_users() -> FLASK_RESPONSE_TYPE:
     if request.method == 'GET':
-        current_pks = [f'user_{int(pk)}' for pk in redis_client.smembers('currentUserPKs')]
+        current_pks = [Users(pk) for pk in redis_client.smembers(CurrentUsers())]
         pipe = redis_client.pipeline()
         for pk in current_pks:
-            pipe.hgetall(pk)
+            pipe.hgetall(Users(pk))
```

Comment:
hm this is actually probably wrong.  doubler Users-ing it.  I think we should try and hook into mypys nominal NewType system to catch this

Reply:
ah but fuck it doesnt understand the metaclassing

Reply:
maybe i can call `cast()` in the __call__ good idea